nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -384,6 +408,46 @@ def forward(self, audio_signal, length, cache_last_channel=None, cache_last_time
         max_audio_length = audio_signal.size(1)
 
         # Create the self-attention and padding masks
+        audio_signal, max_audio_length, pad_mask, att_mask, pos_emb, cache_last_channel_next = self._create_masks(
```

Comment:
What is the benefit of creating this method of _create_masks? Is it going to be used anywhere else?

Reply:
Yes, if we apply subsampling in between the encoder blocks then we need to calculate the masks again because the length of the audio signal will be reduced. So for this purpose, I put it in a separate function.

See line 434. I call this function there again.